lib/Mongo/MongoCollection.php

```diff
@@ -473,15 +535,30 @@ public function findOne(array $query = [], array $fields = [], array $options =
      */
     public function createIndex(array $keys, array $options = [])
     {
-        // Note: this is what the result array should look like
-//        $expected = [
-//            'createdCollectionAutomatically' => true,
-//            'numIndexesBefore' => 1,
-//            'numIndexesAfter' => 2,
-//            'ok' => 1.0
-//        ];
+        if (! is_array($keys) || ! count($keys)) {
+            throw new MongoException('keys cannot be empty');
+        }
+
+        // duplicate
+        $indexes = $this->collection->listIndexes();
+        foreach ($indexes as $index) {
+            if ($index->getKey() === $keys) {
+                throw new \MongoDuplicateKeyException();
```

Comment:
A quick test shows that this exception is not actually thrown:
```
php > $client = new MongoClient();
php > var_dump($client->selectCollection('mongo-php-adapter', 'test')->createIndex(['foo' => 1]));
array(4) {
  'createdCollectionAutomatically' =>
  bool(true)
  'numIndexesBefore' =>
  int(1)
  'numIndexesAfter' =>
  int(2)
  'ok' =>
  double(1)
}
php > var_dump($client->selectCollection('mongo-php-adapter', 'test')->createIndex(['foo' => 1]));
array(5) {
  'createdCollectionAutomatically' =>
  bool(false)
  'numIndexesBefore' =>
  int(2)
  'numIndexesAfter' =>
  int(2)
  'note' =>
  string(25) "all indexes already exist"
  'ok' =>
  double(1)
}
```

There is an exception if the index already exists with different options:
```
php > var_dump($client->selectCollection('mongo-php-adapter', 'test')->createIndex(['foo' => 1], ['unique' => true]));
PHP Warning:  Uncaught exception 'MongoResultException' with message 'localhost:27017: Index with name: foo_1 already exists with different options' in php shell code:1
Stack trace:
#0 php shell code(1): MongoCollection->createIndex(Array, Array)
#1 {main}
  thrown in php shell code on line 1

Warning: Uncaught exception 'MongoResultException' with message 'localhost:27017: Index with name: foo_1 already exists with different options' in php shell code:1
Stack trace:
#0 php shell code(1): MongoCollection->createIndex(Array, Array)
#1 {main}
  thrown in php shell code on line 1
```